### PR TITLE
Add Asn.S.time, which is a choice2 utc_time generalized_time

### PR DESCRIPTION
**Pull request by hannesm**
/cc @reynir please let me know what you think :)

### CODE REVIEW

File: src/asn_combinators.ml [hunk 1]
```diff
@@ -194,3 +194,20 @@ let choice6 a b c d e f =
...
+  let f = function
+    | `C1 t -> t
+    | `C2 t ->
+      if not frac_s && not Ptime.Span.(equal zero (Ptime.frac_s t)) then
+        parse_error "generalized time has fractional seconds"
+      else
+        t
+  and g t =
+    let (y, _, _) = Ptime.to_date t in
+    if y < 2050 then
```
[the remark this answers is not in the document]
Comment (hannesm) [reply src/asn_combinators.ml]: so you mean: if `frac_s` is true, we always need to use GeneralizedTime? So what's the purpose of utc_time then?

[the remark this answers is not in the document]
Comment (reynir) [reply src/asn_combinators.ml]: Yes, or we clarify in the documentation that the time is truncated to seconds when encoding date times within the `utc_time` range.

File: src/asn.mli [hunk 2]
```diff
@@ -305,6 +305,14 @@ module S : sig
 
       Representable years are 1950–2049. *)
 
+  val time : ?frac_s:bool -> unit -> Ptime.t t
+  (** [time ~frac_s ()] is a ASN.1 [choice2 generalized_time utc_time], where
+      [utc_time] is used for years between 1950 and 2049. Otherwise
+      [generalized_time] is used. If [frac_s] is provided and [false] (the
+      default), fractional seconds will lead to errors when decoding, and
+      timestamps are truncated when encoding.
+  *)
```
Comment (reynir) [inline src/asn.mli:314]:
```suggestion
      timestamps are truncated when encoding. When [frac_s] is [true] when
      encoding the timestamps are truncated when within {!utc_time} range only.
  *)
```

**Comment by hannesm** [comment]
I think this PR is the wrong path to take.

**Pull request closed without merging**